### PR TITLE
Fix tri convert upper header

### PR DESCRIPTION
**Pull request by EarthChen**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java [hunk 1]
```diff
@@ -88,27 +101,26 @@ public static void convertAttachment(DefaultHttp2Headers headers,
...
-
+        Map<String, String> needConvertKey = new HashMap<>();
         for (Map.Entry<String, Object> entry : attachments.entrySet()) {
-            final String key = entry.getKey().toLowerCase(Locale.ROOT);
-            if (Http2Headers.PseudoHeaderName.isPseudoHeader(key)) {
-                continue;
+            String key = lruHeaderMap.get(entry.getKey());
+            if (key == null) {
+                final String lowerCaseKey = entry.getKey().toLowerCase(Locale.ROOT);
+                lruHeaderMap.put(entry.getKey(), lowerCaseKey);
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:109]: concrruent issue

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java [hunk 2]
```diff
@@ -88,27 +101,26 @@ public static void convertAttachment(DefaultHttp2Headers headers,
...
+                key = lowerCaseKey;
             }
             if (TripleHeaderEnum.containsExcludeAttachments(key)) {
                 continue;
             }
+            if (needConvertHeaderKey && !key.equals(entry.getKey())) {
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:115]: costly op

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java [hunk 3]
```diff
@@ -44,6 +43,23 @@ public class StreamUtils {
     private static final Base64.Decoder BASE64_DECODER = Base64.getDecoder();
     private static final Base64.Encoder BASE64_ENCODER = Base64.getEncoder().withoutPadding();
 
+
+    private static final int MAX_LRU_HEADER_MAP_SIZE = 1024;
+
+    private static final Map<String, String> lruHeaderMap = new LinkedHashMap<String, String>(MAX_LRU_HEADER_MAP_SIZE, 1f, true) {
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:49]: why add this LRU map ?

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java (hunk 1 above)
Comment (EarthChen) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:109]: The concurrency problem here is ignored and does not affect the final result

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java (hunk 3 above)
Comment (EarthChen) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:49]: reduce tolower

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**